pkg/cmd/cli/cmd/debug.go

```diff
@@ -511,13 +526,21 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*kap
 	}
 	container.Env = newEnv
 
-	if o.AsRoot {
-		if container.SecurityContext == nil {
-			container.SecurityContext = &kapi.SecurityContext{}
-		}
+	if container.SecurityContext == nil {
+		container.SecurityContext = &kapi.SecurityContext{}
+	}
+
+	switch {
+	case o.AsNonRoot:
+		b := true
+		container.SecurityContext.RunAsNonRoot = &b
+	case o.AsUser != -1:
+		container.SecurityContext.RunAsUser = &o.AsUser
 		container.SecurityContext.RunAsNonRoot = nil
+	default:
 		zero := int64(0)
 		container.SecurityContext.RunAsUser = &zero
```

Comment:
Yeah, refactored to be simpler right before pushing and didn't change the default case.